### ISSUE TITLE
Yarn cannot properly install Bourbon

### ISSUE THREAD

**Issue by leemcd56** [thoughtbot/bourbon#990]
### Expected behavior
`yarn add bourbon` should add bourbon to package.json and install it within node_modules.

### Actual behavior
`yarn add bourbon` does add bourbon to package.json, however installs it without `app/assets/stylesheets`. However, bourbon DOES install properly through `npm install`. When running `yarn upgrade` afterwards, the same results from `yarn add` occur.

### Steps to reproduce behavior
In any project, run `yarn add bourbon`.

### Bourbon version
4.2.7

### Environment info
macOS Sierra 10.12.3 Beta (16D17a)
node 6.9.4
npm 4.0.5
yarn 0.19.1

### Screenshots (if relevant)
<img width="577" alt="screen shot 2017-01-24 at 1 44 39 pm" src="https://cloud.githubusercontent.com/assets/1885663/22263850/7e55400e-e23b-11e6-99bd-f8521d3e44c9.png">
<img width="962" alt="screen shot 2017-01-24 at 1 45 05 pm" src="https://cloud.githubusercontent.com/assets/1885663/22263862/863b5542-e23b-11e6-82bc-e55eaec96aa2.png">
<img width="359" alt="screen shot 2017-01-24 at 1 44 20 pm" src="https://cloud.githubusercontent.com/assets/1885663/22263867/897a48b2-e23b-11e6-9029-42ab35de01b4.png">
<img width="747" alt="screen shot 2017-01-24 at 1 46 01 pm" src="https://cloud.githubusercontent.com/assets/1885663/22263872/8cb9f7a2-e23b-11e6-820a-fd0122d50271.png">

**Comment by Toddses** [comment]
FWIW, I just tested this out and bourbon installed correctly. I'm using the same versions as you, except for NPM (mine is 3.10.10).

**Comment by leemcd56** [comment]
This is rather curious. Strangely enough, this open happens with two packages, `open-color` and `bourbon`, both of which properly install in `npm`, but not `yarn`.

**Comment by Toddses** [comment]
Tested it out with `open-color` - no issues on that one either.

**Comment by leemcd56** [comment]
I wonder if this is a `yarn` issue?

**Comment by Toddses** [comment]
Or a MacOS Beta issue? I am using 10.12.2.

For giggles, I tried a number of scenarios and `yarn` installed/uninstalled all packages with no issues. Also it was insanely faster - I see it took 20+ seconds for you to install bourbon, while I was able to install in <1 second. Could the installation be timing out? Do any of the other installed packages depend on bourbon? Can you create a new project, `npm init` and `yarn install bourbon` with no issues (ie test it out outside of the context of your current project).

**Comment by leemcd56** [comment]
In a new project, `yarn add bourbon` adds Bourbon with no issue. It appears this is only specific to [roots/sage](https://github.com/roots/sage).

**Comment by Toddses** [comment]
I cloned sage and installed the dependencies with `yarn`. then added bourbon. bourbon installed fine, though i did not have the fifth step there, `Cleaning modules...`.

**Comment by leemcd56** [comment]
¯\_(ツ)_/¯ Could be a Sierra beta issue, or something to do with the yarn installation itself.

**Comment by tysongach** [comment]
Have you filed an issue with Yarn? They [claim in their documentation](https://yarnpkg.com/en/docs/migrating-from-npm) that “Yarn can consume the same `package.json` format as npm, and can install any package from the npm registry.” To be thorough, I just tested install Bourbon through npm and all was good.

With that, the problem seems to be an issue with your specific environment and Yarn, and not with Bourbon.

**Comment by leemcd56** [comment]
It is very odd behavior for Bourbon to install through Yarn just fine in other projects, but this issue occur only on *this* specific project. Only Node modules I have added in that don't already come with the project seem to fail. I will certainly file an issue with Yarn.

**Comment by tysongach** [comment]
@leemcd56 Indeed some very strange behavior. Please do follow up with and reference your findings!

**Comment by leemcd56** [comment]
I found [this issue](https://github.com/yarnpkg/yarn/issues/785) that seems to be related. As @Toddses pointed out, the difference between my install process and his was the `Cleaning modules...` step.

So, I did some research on that. Apparently, the `yarn clean` command is being automatically run on the `yarn install` and `yarn add` commands. It generates a `.yarnclean` file, which lists out directories and files Yarn will remove from modules. Guess what directory is on the last?

<img width="355" alt="screen shot 2017-01-24 at 7 33 04 pm" src="https://cloud.githubusercontent.com/assets/1885663/22274409/f8f5dd66-e26b-11e6-81fd-a26263474856.png">

Commenting `assets` out and running `yarn add bourbon` allowed it to properly install. Technically, this is a Bourbon issue, but unintentionally. If the directory was named `src` instead, it'd probably prevent this in the future.
